cns/service/main.go

```diff
@@ -312,8 +313,14 @@ func printVersion() {
 	fmt.Printf("Version %v\n", version)
 }
 
+// NodeInterrogator is functionality necessary to read information about nodes.
+// It is intended to be strictly read-only.
```

Comment:
Yeah, the intention is to set some opinions about what this set of behavior means. If you have some methods which aren't about digging information out of Nodes, then it might belong in another interface.